File: 530. Minimum Absolute Difference in BST.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int result = INT_MAX;
    TreeNode* pre = nullptr;

    void traversal(TreeNode* root) {
        if(root == nullptr) {
            return;
        }
        traversal(root->left);
        if(pre != nullptr && root->val - pre->val < result) {
            result = root->val - pre->val;
        }
        pre = root;
        traversal(root->right);
    }

    int getMinimumDifference(TreeNode* root) {
        traversal(root);
        return result;
    }
};

File: 530. Minimum Absolute Difference in BST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int result = Integer.MAX_VALUE;
    TreeNode pre = null;

    void traversal(TreeNode root) {
        if(root == null) {
            return;
        }
        traversal(root.left);
        if(pre != null && root.val - pre.val < result) {
            result = root.val - pre.val;
        }
        pre = root;
        traversal(root.right);
    }

    public int getMinimumDifference(TreeNode root) {
        traversal(root);
        return result;
    }
}